### DISCUSSION TITLE
Using vite for laravel mails, Vite::content in dev mode

### DISCUSSION THREAD

**Discussion by Coding-Kiwi** [laravel/framework#51721] [Q&A]
I use scss in my app and to have my mails sent by laravel themed too, I used this

```js
mix.sass('resources/scss/mail/default.scss', 'public/css/mail')

mix.webpackConfig({
    plugins: [
        new FileManagerPlugin({
            events: {
                onEnd: {
                    copy: [{ source: './public/css/mail', destination: './resources/views/vendor/mail/html/themes' }],
                },
            }
        }),
  ]
});
```

to build and copy the css to the required location.

Now I want to use vite and as stated in the docs there is a way to include raw content https://laravel.com/docs/11.x/vite#inline-assets

```blade
@use('Illuminate\Support\Facades\Vite')
 
<!doctype html>
<head>
    {{-- ... --}}
 
    <style>
        {!! Vite::content('resources/scss/mail/default.scss') !!}
    </style>
</head>
```

This only works when the manifest.json is in the public/build directory which is not the case when running vite in dev mode. This is only the case when I manually run the production build command.

Is there a better way to do this?

**Comment by wizzymore** [comment]
While you are working on your emails you can use the vite watch mode which rebuilds you js and css on changes instead of serving them via websockets.

You can also create a separate vite config only for emails and pass it as a parameter on a new npm command and use that in watch mode for email. You can run multiple npm commands using the package `concurrently` so you can have the best of both worlds and don't keep track of what servers you started.

Example:
```json
...
"scripts": {
    "dev": "concurrently \"npm run vite:dev\" \"npm run vite:mail\"",
    "vite:dev": "vite",
    "vite:mail": "vite build -c vite.mail.config.js --watch",
    ...,
}
...
```
